graphql/schema/schemagen.go

```diff
@@ -394,7 +394,10 @@ func NewHandler(input string, apolloServiceQuery bool) (Handler, error) {
 	}
 
 	metaInfo.extraCorsHeaders = getAllowedHeaders(sch, defns, authHeader)
-	dgSchema := genDgSchema(sch, typesToComplete, providesFieldsMap)
+	dgSchema, gqlErr := genDgSchema(sch, typesToComplete, providesFieldsMap)
+	if gqlErr != nil {
```

Comment:
I believe the errors are returned in case the provided schema is not valid. Why isn't the validation done as part of postGQLValidation. You may do all the validation inside the function dgraphDirectiveValidation